### DISCUSSION TITLE
Implementing Pintura within a NiceGui application

### DISCUSSION THREAD

**Discussion by whoulden** [zauberzeug/nicegui#3946] [Ideas / Feature Requests]
First up I'd like to thank you for creating such a powerful framework. I've been using it now for a few months and whilst I'm not an advanced Python or front-end developer, I've found your tech, documentation, examples and support to be superb - you (collectively) must be very proud of what you've achieved to date.

I'm building a generative AI application for online assessment designs - it's a pretty reasonable sized piece of work and to date NiceGui has been able to make this come together quite quickly.

Here's my question - as part of assessment generation, I'm creating images for stimuli and I'd like to be able provide my users with a tool to edit those generated images. For that I'm thinking of using Pintura. 

I'm having a crack at working out if it can be implemented, and I thought I'd ask you if you had implemented image editor support before, and if so, what you have used.

Any thoughts or guidance on getting Pintura up and going would also be greatly appreciated.

**Comment by falkoschindler** [comment]
Hi @whoulden,

Thank you for your kind words and for supporting this project! Your contribution is very welcome.

We haven't heard of Pintura yet, but it looks pretty awesome. Even though we're not sure if NiceGUI should ship with such a powerful image editor included, it might make a great [example](https://github.com/zauberzeug/nicegui/tree/main/examples/). However, their [license](https://pqina.nl/pintura/license/) seems to be rather restrictive, so that we might not be allowed to distribute Pintura as part of NiceGUI. Thus, the example would need to include instructions how to resolve the dependency manually.
Besides that, I couldn't immediately see how to use it in an environment like a custom NiceGUI component. Downloading their [ZIP file](https://pqina.nl/pintura/install/#zip-panel) returns `{"error": "Not found"}`.

I'm still hopeful that a Pintura example is possible. But it doesn't feel trivial to setup.

**Reply by whoulden** [reply]
Thanks Falko

Sorry about the delay in writing back, it ended up being relatively easy to setup as a javascript integration and works quite nicely, including a post back to save the image.

You're right about the license - its certainly restrictive but given our needs for reasonable editing capabilities, we are likely to license it.

It certainly was a good exercise in understanding how such tools could be used outside of the NiceGui/Vue component architecture.
